p2p/protocol/autonatv2/client.go

```diff
@@ -248,6 +250,12 @@ func newDialRequest(reqs []Request, nonce uint64) pb.Message {
 
 // handleDialBack receives the nonce on the dial-back stream
 func (ac *client) handleDialBack(s network.Stream) {
+	defer func() {
```

Comment:
Do we need to ensure that the stream has been closed?

Reply:
Yes. We should reset the stream on panic.